### DISCUSSION TITLE
On Row Click will not display related form in other screen - MySQL server

### DISCUSSION THREAD

**Discussion by ebrodeur** [Budibase/budibase#11490] [Help] [answered]
Hello. I'm new to Budibase (self-hosted) and have spent a lot of time kicking the tires. I have an app connected to a self-hosted MySQL server (MariaDB).

The app connects to the database OK.

I have a screen which shows me a list of People. When a row is clicked (the row or a button, doesn't matter to me), I want to open the related record in a new screen where I have placed fields to my liking (using Repeaters and Text Fields, not Table Block).

Based on the forum posts I've read, this should be easy-peasy, however none of those solutions work for me.

I've read posts where Navigate To should be "/people/:id". I've read other posts which it should be "/people/:id/{{ People Table.people._id }}".

On "/people/:id/" the Data Source filter is set to "_id equals binding {{ URL.id }}".

If I click on "John Doe" in "/people" it pulls up whatever record was last viewed in "/people/:id" - not John Doe.

I've tried all variations (._id, .id, other binding names based on the component's context, etc.) and cannot get anything to work.

The app templates in Budibase work using the above solutions, however they're using the built-in BB database. These don't seem to work with MySQL.

Can someone tell me what I'm doing wrong?

Thank you.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @ebrodeur,

What version of budibase are you running?

I was able to do this relatively easily below is a list of the entire process i took.

1. Create a MySQL table using the below script;
```
CREATE TABLE persons (
    person_id INT PRIMARY KEY,
    first_name VARCHAR(50),
    last_name VARCHAR(50),
    date_of_birth DATE,
    gender VARCHAR(10),
    email VARCHAR(100),
    phone_number VARCHAR(20),
    address VARCHAR(200),
    city VARCHAR(50),
    state VARCHAR(50),
    postal_code VARCHAR(20)
);
```
2. Created a screen called `/persons`
3. Created an additional screen called `/persons/:id`
4. Added a Data provider + table to a screen.
5. Connected the Data provider to the persons table created earlier.
6. Click the on click row button and add a `navigate to` action.
![Screenshot 2023-08-10 at 15 28 16](https://github.com/Budibase/budibase/assets/126772285/143a4f37-1de5-48cc-a84d-d302b4476d84)
7. Pass in the Clicked row._id replacing the :id part of the url in the previous created `/persons/:id` screen
8. On the persons/:id screen add a form block
9. Set the type to update
10. Set the row id to `{{ URL.id }}`
11. Optionally you can customise by whatever fields you wish using the `All fields button`
12. Additionally you can eject the component the component to see how everything works in a more broken down format.
![Screenshot 2023-08-10 at 15 32 56](https://github.com/Budibase/budibase/assets/126772285/ea8351ec-177a-4021-b23b-001a3af63664)

This should be everything setup for you. 

**Output**
![b4744e9c385b45458da812db27a457bb](https://github.com/Budibase/budibase/assets/126772285/48747f70-7c2f-419e-94f8-922c24efd64f)

I hope this helps!

**Comment by ebrodeur** [comment]
Hi @ConorWebb96 and thanks for the reply. I'm using version 2.8.30.

Your solution works. This is the first I've seen about {{ Clicked row.id }}.

However, for my desired use case, I want to manually arrange fields on the screen. The "normal" Form Block doesn't allow me to use Field Groups inside of it.

When I create a Field Group, it requires being inside of a Form Block, however that Form Block doesn't allow for choosing Row ID as your example shows.

This is an example of the form I want to build. See the red boxes.

<img width="1613" alt="1_data provider" src="https://github.com/Budibase/budibase/assets/401552/5768c39a-2b1e-4677-97d1-05ee6a5c5d46">

For the Data Provider, I've set the Filter to {{ URL.id }} as I've seen in other forum posts.

<img width="1617" alt="2_data provider filter" src="https://github.com/Budibase/budibase/assets/401552/805fb88c-7581-45b0-8e44-78b8cc2bcb51">

On the "New Form" component what schema am I supposed to use? No matter what I choose here, the {{ Clicked row.id }} value from the main /people table does not pass through.

<img width="1605" alt="Screenshot 2023-08-11 at 11 34 58 AM" src="https://github.com/Budibase/budibase/assets/401552/0f963b9a-3514-48fd-b329-3dbf38006e4e">

Do I need to use a different combination of components to make this work? Additional code?

Thanks for your help!

**Reply by melohagan** [reply]
Hey @ebrodeur 

If you had it working with the **Form Block**, the easiest thing to do would have been to [Eject](https://docs.budibase.com/docs/blocks#ejecting-blocks) the block to get the component parts. 

You can see on that docs page that a [Repeater](https://docs.budibase.com/docs/repeater) component is added, which is necessary to pull row data from a data provider.

**Reply by ebrodeur** [reply]
Hi Melohagan and thanks for the reply.

Repeater block got it working for me. I will try the Eject function in the near future.

The documentation is a deluge of information and a bit overwhelming for a first-timer. The Repeater page describes what solved my issue, however it wasn't jumping out at me as the solution. Perhaps some specific use-case examples would be helpful for other folks starting out.

Thanks again for the help! Much appreciated.

**Reply by melohagan** [reply]
Hey @ebrodeur Glad you got it working, and thanks for the feedback! I am working my way through some specific use-case tutorials at the moment.

**Comment by ebrodeur** [comment]
That's great to hear about the tutorials.

I haven't used the built-in BB database so if it exposes component features differently than an external data source (ie: SQL server), it would be good to have the tutorial cover both.

Thanks again.
